static/js/stats.js

```diff
@@ -179,16 +155,201 @@ function plot(tIssued, tActive, tFqdn, tRegDom, tPctTLSAvg) {
+  let sortedDates = Object.keys(dateToCountryOSPageloadData);
+  sortedDates.sort();
+  for (let datestamp of sortedDates) {
+    // See https://bugzilla.mozilla.org/show_bug.cgi?id=1414839#c19 for
+    // parameter details
+    let totalNormalizedPageloads = 0.0;
+
+    // Preprocess to sum all normalized_pageloads for included rows
+    for (let country in dateToCountryOSPageloadData[datestamp]) {
+      for (let os in dateToCountryOSPageloadData[datestamp][country]) {
+        if (includeInHttpsAnalysis(os, country)) {
+          let row = dateToCountryOSPageloadData[datestamp][country][os];
+          totalNormalizedPageloads += parseFloat(row.normalized_pageloads);
+        }
+      }
+    }
+
+    // Now derive the reporting ratio and secure pageload ratio
+    let totalSecurePageloadRatio = 0.0;
+
+    for (let country in dateToCountryOSPageloadData[datestamp]) {
+      for (let os in dateToCountryOSPageloadData[datestamp][country]) {
+        if (includeInHttpsAnalysis(os, country)) {
+          let row = dateToCountryOSPageloadData[datestamp][country][os];
+          let dimensionNormalizedPageloads = parseFloat(row.normalized_pageloads)
+                                               / totalNormalizedPageloads;
+          totalSecurePageloadRatio += parseFloat(row.ratio)
+                                         * dimensionNormalizedPageloads;
+        }
+      }
+    }
+
+    let securePageloadPercentage = movingAvg(stackMovingAverage,
+                                             totalSecurePageloadRatio * 100);
+    insertPoint(traceObj, datestamp, securePageloadPercentage);
+  }
+}
+
+function historicalHttpsCsvListener(responseText) {
+  parse_delim(responseText, ',', function(row){
+    let datestamp = row[0];
+    let pageloads = row[1];
+    if (!dateFormat.test(datestamp)) {
+      return;
+    }
+
+    gHttpsData.dateToHistoricalPageloadData[datestamp] = pageloads;
+  });
+}
+
+function importHistoricalGlobalData(traceObj, stackMovingAvg) {
+  let sortedDates = Object.keys(gHttpsData.dateToHistoricalPageloadData);
+  sortedDates.sort();
+  for (let datestamp of sortedDates) {
+    let pageloadPercent = gHttpsData.dateToHistoricalPageloadData[datestamp];
+    let securePageloadPercentage = movingAvg(stackMovingAvg,
+                                             pageloadPercent);
+    insertPoint(traceObj, datestamp, securePageloadPercentage);
+  }
+}
+
+// Firefox telemetry (HTTP_PAGELOAD_IS_SSL) over time
+function httpsPlot() {
+  let traces = [];
+
+  {
+    let traceObj = { type: "scatter", x:[], y:[], name: "Global users" }
+    let stackMovingAvg = [];
+    importHistoricalGlobalData(traceObj, stackMovingAvg);
+    httpsDerivePageloadsFromNormalizedData(traceObj, () => {
+      return true;
+    }, stackMovingAvg);
+    traces.push(traceObj);
+  }
+  {
+    let traceObj = { type: "scatter", x:[], y:[], name: "Germany users" }
+    httpsDerivePageloadsFromNormalizedData(traceObj, (os, country) => {
```

Comment:
I'm pretty sure this could be made more efficient than calling the heavy `httpsDerivePageloadsFromNormalizedData` method multiple times, but I wanted to go for clarity rather than speed.